utils/test_results.py

```diff
@@ -1,101 +1,315 @@
 import datetime as dt
-from dataclasses import dataclass
+from math import floor
 
-from django.contrib.postgres.aggregates import ArrayAgg
 from django.db.models import (
+    Aggregate,
     Avg,
-    Case,
     F,
     FloatField,
     Func,
-    IntegerField,
     Max,
+    OuterRef,
     Q,
     QuerySet,
-    Value,
-    When,
+    Subquery,
+    Sum,
+)
+from django.db.models.functions import Cast
+from shared.django_apps.core.models import Repository
+from shared.django_apps.reports.models import (
+    DailyTestRollup,
+    Flake,
 )
-from django.db.models.functions import Coalesce
-from shared.django_apps.reports.models import TestInstance
 
 thirty_days_ago = dt.datetime.now(dt.UTC) - dt.timedelta(days=30)
 
+SLOW_TEST_PERCENTILE = 95
+
 
-@dataclass
-class TestResultsAggregation:
-    failure_rate: float | None
-    commits_where_fail: list[str] | None
-    average_duration: float | None
+def slow_test_threshold(num_tests: int):
+    threshold = floor(num_tests * (100 - SLOW_TEST_PERCENTILE) * 0.01)
+    if threshold == 0:
+        if num_tests < (1 / ((100 - SLOW_TEST_PERCENTILE) * 0.01)):
```

Comment:
Is this line needed? When would this statement be false if we're taking the floor already on line 30